docs/content/preview/yugabyte-platform/security/enable-encryption-at-rest.md

```diff
@@ -56,59 +49,85 @@ You can verify that encryption at rest has been successfully configured as follo
+1. In the **Manage Encryption at Rest** dialog, toggle **Enable Encryption at Rest for this Universe**.
 
    When the encryption is enabled, the **Key Management Service Config** option appears.
 
-5. Select your KSM configuration from the **Key Management Service Config** list. The list displays only preconfigured KMS configurations. If you need to create one, see [Create a KMS configuration](../create-kms-config/).
+1. Select your KSM configuration from the **Key Management Service Config** list. The list displays only preconfigured KMS configurations. If you need to create one, see [Create a KMS configuration](../create-kms-config/).
 
-6. Click **Submit**.
+1. Click **Apply**.
 
-7. Verify that the `EnableEncryptionAtRest` task completed successfully.
+1. Verify that the `EnableEncryptionAtRest` task completed successfully.
 
 ## Back up and restore data from an encrypted at rest universe
 
-When you back up and restore universe data with encryption at rest enabled, YugabyteDB Anywhere requires a KMS configuration to manage backing up and restoring encrypted universe data. Because of the possibility that you will need to restore data to a different universe that might have a different universe key, YugabyteDB Anywhere ensures that all encrypted backups include a metadata file (that includes a list of key IDs in the source's universe key registry). When restoring your universe data, YugabyteDB Anywhere uses the selected KMS configuration to consume the universe key ID and then retrieve the universe key value for each key ID in the metadata file. Each of these keys are then sent to the destination universe to augment or build the universe key registry there.
+When you back up and restore universe data with encryption at rest enabled, YugabyteDB Anywhere requires a KMS configuration to manage backing up and restoring encrypted universe data. Because of the possibility that you will need to restore data to a different universe that might have a different master key, YugabyteDB Anywhere ensures that all encrypted backups include a metadata file. The file includes a list of key IDs in the source's master key registry.
+
+When restoring an encrypted backup to a universe, Yugabyte Anywhere detects the correct KMS configuration used to encrypt the backup. The KMS configuration must be available in the YugabyteDB Anywhere account.
+
+When restoring your universe data, YugabyteDB Anywhere uses the selected KMS configuration to consume the master key ID and then retrieve the master key value for each key ID in the metadata file. Each of these keys are then sent to the destination universe to augment or build the universe key registry there.
+
+## Rotate the master keys
+
+As part of envelope encryption, the universe keys are protected by master keys. The master key resides in the KMS of your choosing and is used to encrypt and decrypt the universe keys as needed.
```

Comment:
what is envelope encryption? I got the idea after googling it , but should we provide a link to it or can we assume that a user would be familiar with it?

Reply:
added in 1st paragraph on page